# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/RohanRamchandani/FlyRank-ML-Week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I chose this lane because it builds directly on the workflow I already practiced in Notebooks 1 and 2 — comparing a transparent hand-written rule against a learned model, and reading exactly what that model found instead of trusting a black-box score. Lane 2 also produces a concrete, useful output: a ranked list of pages a content team should review first, with clear reasons attached to each recommendation. That combination of interpretability and real decision-support is what makes this lane the most natural fit for me to go deeper on.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**Decision this improves:** which pages a content/SEO reviewer should look at first when
deciding whether to refresh, expand, protect, prune, or monitor a piece of content.

**Who acts on it:** a content or SEO team with limited review capacity — they can't manually
audit every page, so they need a ranked shortlist of the highest-priority candidates.

**Cost of a wrong call:**
- A false positive (flagging a healthy page as needing review) wastes a reviewer's limited
  time on a page that didn't need attention.
- A false negative (missing a page that is genuinely declining) means a real traffic/visibility
  problem goes unnoticed and unaddressed, potentially costing the client organic traffic and
  revenue over time.

Because review capacity is limited, the ranking itself matters as much as raw accuracy —
getting the top of the list right is more valuable than getting every single page right.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

total = len(df)
declining = (df["trend_direction"] == "down").sum()
print(f"Total pages: {total}")
print(f"Declining pages: {declining} ({declining/total:.1%} of all pages)")

stale_visible = df[(df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)]
print(f"\nStale AND still visible (500+ impressions): {len(stale_visible)} pages")
print(f"({len(stale_visible)/total:.1%} of all pages — strong refresh candidates)")

declining_with_demand = df[(df["trend_direction"] == "down") & (df["impressions_90d"] >= 100)]
print(f"\nDeclining pages with real demand (100+ impressions): {len(declining_with_demand)} pages")

Total pages: 30000
Declining pages: 16262 (54.2% of all pages)

Stale AND still visible (500+ impressions): 17 pages
(0.1% of all pages — strong refresh candidates)

Declining pages with real demand (100+ impressions): 13152 pages


Looking at the starter dataset of 30,000 pages: 16,262 pages (54.2%) are currently declining —
a substantial majority of the content inventory. More importantly, 13,152 of those declining
pages (43.8% of all pages) still have real demand (100+ impressions), meaning the decline isn't
just noise on dead pages — most of these are actively visible pages losing ground while people
are still searching for them. This is exactly the kind of gap a review queue should catch early.

Only 17 pages (0.1%) are both stale (180+ days since last update) and still highly visible
(500+ impressions) — a small but very high-confidence slice of "obvious neglect" candidates,
useful as a sanity-check baseline rather than the whole opportunity.

Together, these numbers show the real opportunity in this lane: a large volume of declining,
still-visible pages that a limited review team could realistically never catch manually —
which is exactly the prioritization problem Lane 2 is built to solve.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [17]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print(df.columns.tolist())

['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


**What my work can say:**
- Which pages are *observed* to be declining, stale, or under-performing relative to their
  peers, based on the data available.
- A *directional* ranking of which pages are more or less likely to be worth reviewing first.
- *Decision-support*: a prioritized shortlist for a human reviewer, not an automated verdict.
- That a page's decline is *associated* with certain signals (e.g. staleness, position, CTR) —
  not that those signals *caused* the decline.

**What my work cannot say:**
- That refreshing a flagged page *will cause* it to recover — that would require a controlled
  experiment (e.g. before/after with a valid causal design), which this data alone cannot provide.
- That any result here reflects or predicts Google's actual ranking algorithm.
- That a high or low score is a guarantee, rather than a probability-weighted priority.
- Anything that names or implies a specific client, URL, or query — all outputs must stay
  pseudonymized and aggregate.
- That the 54.2% decline rate observed in this starter dataset represents typical real-world
  conditions — this is a small teaching sample, not a general benchmark.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.